arbnode/node.go

```diff
@@ -770,6 +787,51 @@ func createNodeImpl(
 	}, nil
 }
 
+func FindBlockContainingBatchCount(ctx context.Context, bridgeAddress common.Address, l1Client *ethclient.Client, parentChainAssertionBlock uint64, batchCount uint64) (uint64, error) {
+	bridge, err := bridgegen.NewIBridge(bridgeAddress, l1Client)
+	if err != nil {
+		return 0, err
+	}
+	high := parentChainAssertionBlock
+	low := uint64(0)
+	if high > 100 {
+		low = high - 100
+	}
+	// Reduce high and low by 100 until lowNode.InboxMaxCount < batchCount
+	// This will give us a range (low to high) of blocks that contain the batch count.
+	for low > 0 {
+		lowCount, err := bridge.SequencerMessageCount(&bind.CallOpts{Context: ctx, BlockNumber: new(big.Int).SetUint64(low)})
+		if err != nil {
+			return 0, err
+		}
+		if lowCount.Uint64() > batchCount {
+			high = low
+			if low > 100 {
```

Comment:
This could be millions of blocks in the past.
Instead of a fixed value "100" start with a "range := 100", then if we need to go further back do
```
  high = low
  low = low - range
  range = range * 2
```
(not exact, e.g. need to see we're never going beyond block 0 etc)